tests/python_package_test/test_dask.py

```diff
@@ -432,9 +434,8 @@ def test_regressor(output, client):
     p2 = local_regressor.predict(X)
 
     # Scores should be the same
-    if not output.startswith('dataframe'):
-        assert_eq(s1, s2, atol=.01)
-        assert_eq(s1, s1_local, atol=.003)
+    assert_eq(s1, s2, atol=.01)
+    assert_eq(s1, s1_local, atol=0.003)
```

Comment:
when you tested, did you try running these tests multiple times in a row? I'd like to know how close these tolerances are to the maximum difference that you saw. It's good to push these tolerances down to catch big differences between distributed training and non-distributed training, but not so close to the edge that the tests might fail sometimes due to randomness in how the chunks are distributed across the workers.

Reply:
For the test above the maximum absolute difference I observed was 0.0025, so 0.01 should be fine. For this one I believe we can even remove the atol, since this checks the score for the same model, doesn't it?

Reply:
great, thank you. Yes, I think you're right. `s1` and `s1_local` should be identical since they're from the same model.

Reply:
I played around with the checks for the predictions and the target and found that for `atol=50` we could go to `rtol=0.5` (the maximum observed was 0.25). But maybe we could instead check `p1` vs `p2`? They're very similar so maybe that'd be a better check and could go very low in the tolerances as well. Also I switched the order in the comparison for this one because `assert_eq(a, b)` checks `abs(a - b) <= atol + rtol * abs(b)`, so I think it makes more sense that `b = y` in this case. What do you think?

Reply:
ok I'm a little confused by your notation, sorry. Are you referring to this change specifically?

![image](https://user-images.githubusercontent.com/7608904/112663396-c5996d00-8e26-11eb-92b6-2051acdb0297.png)

To be honest, I don't understand what you mean. Can you give a small reproducible example that shows `assert_eq()` passing or failing based only on the order that you pass things in?

To me, it looks like the checks in that function are symmetric: https://github.com/dask/dask/blob/913a192fd3b35696883bdd4c4eeba2418b946b3a/dask/dataframe/utils.py#L841-L847.

Reply:
Haha, I'm sorry. Yes, I meant those checks exactly. The asymmetric part is in https://github.com/dask/dask/blob/913a192fd3b35696883bdd4c4eeba2418b946b3a/dask/dataframe/utils.py#L847, which calls [`np.allclose`](https://github.com/numpy/numpy/blob/fb215c76967739268de71aa4bda55dd1b062bc2e/numpy/core/numeric.py#L2256), and [`np.isclose`](https://github.com/numpy/numpy/blob/fb215c76967739268de71aa4bda55dd1b062bc2e/numpy/core/numeric.py#L2346)

 Here's an example:
```python
import numpy as np

a = 100
b = 120
np.allclose(a, b, rtol=0.1, atol=9)  # (20 - 9) / 0.1 = 110 < 120 (True)
np.allclose(b, a, rtol=0.1, atol=9)  # (20 - 9) / 0.1 = 110 < 100 (False)
```

Reply:
:exploding_head: I had no idea that these were not symmetric! Thanks for the example. I see now that there's a note about this in the `np.isclose()` docs:

> Unlike the built-in `math.isclose`, the above equation is not symmetric
    in `a` and `b` -- it assumes `b` is the reference value -- so that
    `isclose(a, b)` might be different from `isclose(b, a)`

Ok, since this is checking that we get a "good enough" model result, I agree with the change to treat `y` (the actual value of the target) as the "reference value". Thanks for the attention to detail.

Reply:
also now I know @jmoralez and @jose-moralez are the same person :stuck_out_tongue:

Reply:
haha, yes. This is the account I use at work. I usually contribute on my free time using my personal account, but feel free to ping me on this one at work hours.